### ISSUE TITLE
e2e: Add tests for common user account flows

### ISSUE THREAD

**Issue by Parkreiner** [coder/coder#13199]
Spinning this off into a milestone issue per @Kira-Pilot's recommendation.
This is spurred by issue #13130

## Problem
- As we keep using the Go server to seed more and more metadata directly into the HTML file, we have an increased risk of drift between our JS-based unit tests, and anything that uses the Go server, because the JS tests will never have access to that metadata, and won't ever be able to test for it

## Proposed solution
- To that end, we should have our E2E tests go through core user account flows (e.g., signing out), just to make sure we don't kick up any runtime errors in the UI

## Notes
- I have [the beginnings of something here](13185), but I'm not sure if there's anything else critical that needs to be added (particularly for the different types of logins we support)

**Comment by BrunoQuaresma** [comment]
I'm missing what core user account flows are because it is VERY open.

**Comment by Parkreiner** [comment]
Yeah, sorry – going to refine the requirements for this later on today
